src/runtime/security/kms_client.h

```diff
@@ -0,0 +1,72 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "absl/strings/str_split.h"
+#include "utils/errors.h"
+
+namespace dsn {
+namespace security {
+// A class to generate encryption_key from kms for writing file which implemented based on http
+// client
+// This class is not thread-safe. Thus maintain one instance for each thread.
+//
+// Example of using Kms client:
+// --------------------------------------------------------
+// Generate and get the Encrypted Encryption Key(eek),Initialization Vector(iv),Key Version from
+// kms:
+// std::string encryption_key;
+// std::string iv;
+// std::string key_version;
+// err = GenerateEncryptionKey(key_name, &encryption_key, &iv, &key_version);
+//
+// Get the Decrypted Encryption Key(dek) back from kms. The eek,iv,kv need generated from kms:
+// std::string decryption_key;
+// err = DecryptEncryptionKey(encryption_key, iv, key_version, &decrypted_key);
+class KMSClient
+{
+public:
+    KMSClient(const std::string kms_url, std::string cluster_key_name)
+        : kms_urls_(::absl::StrSplit(kms_url, ",", ::absl::SkipEmpty())),
```

Comment:
Avoid to call functions in initialize list, you can call the StrSplit when construct KMSClient objects.